### PR TITLE
[citrix_adc] Troubleshoot Date/Time parsing

### PR DESCRIPTION
**Pull request by jrmolin**
## Proposed commit message

[citrix_adc] Expand date/time processing
* Some pipeline errors found where timestamps had trailing whitespace
* Custom timestamp format parsing was not setting `@timestamp` appropriately
* Other formats seen by users that were not handled by grok lines
* Sanitized samples added to the test cases
* Grok patterns updated and added to
* Tests passing

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/citrix_adc/data_stream/log/elasticsearch/ingest_pipeline/tcp_and_acl_feature.yml [hunk 1]
```diff
@@ -6,13 +6,51 @@ processors:
...
         - '%{GREEDYDATA:citrix_adc.log.message}'
+      pattern_definitions:
+        NAT_IP_AND_PORT: 'NatIP %{IP:citrix_adc.log.nat.ip}:%{INT:citrix_adc.log.nat.port}'
+        SOURCE_IP_AND_PORT: 'Source %{IP:citrix_adc.log.source.ip}:%{INT:citrix_adc.log.source.port}'
+        DEST_IP_AND_PORT: 'Destination %{IP:citrix_adc.log.destination.ip}:%{INT:citrix_adc.log.destination.port}'
+        TRANSLATED_DEST_IP_AND_PORT: 'Destination %{IP:citrix_adc.log.translated_destination.ip}:%{INT:citrix_adc.log.translated_destination.port}'
+        VSERVER_IP_AND_PORT: 'Vserver %{IP:citrix_adc.log.vserver.ip}:%{INT:citrix_adc.log.vserver.port}'
+        TOTAL_BYTES_SEND: 'Total_bytes_send %{INT:citrix_adc.log.total_bytes_send:long}'
+        TOTAL_BYTES_RECV: 'Total_bytes_recv %{INT:citrix_adc.log.total_bytes_received:long}'
+        DELINK_TIME: '%{DATA:citrix_adc.log.delink_time}'
```
[the remark this answers is not in the document]
Comment (jrmolin) [reply packages/citrix_adc/data_stream/log/elasticsearch/ingest_pipeline/tcp_and_acl_feature.yml:27]: So the timestamp changes were partially caused by grok updates. There was a lot of duplication in the grok patterns, and it was really hard to keep them all straight. Also, the ordering was causing issues. I separated out patterns to make it easier to keep track of the updates. I can revert it, but it was really helpful.

File: packages/citrix_adc/data_stream/log/_dev/test/pipeline/test-citrix-waf-native.log [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 Oct 6 14:03:23 <local0.info> 81.2.69.144 10/06/2014:14:03:23 GMT ns1 0-PPE-0 : TCP OTHERCONN_DELINK 4471 0 : Source 192.168.1.100:12345 - Vserver 1.128.0.0:443 - NatIP 1.128.0.0:1024 - Destination 1.128.0.0:80 - Delink Time 2023-04-01T12:00:00Z GMT Total_bytes_send 51200 - Total_bytes_recv 102400
 Oct 6 14:03:23 <local0.info> 81.2.69.144 10/06/2014:14:03:23 GMT ns1 0-PPE-0 : TCP NAT_CONN_DELINK 4471 0 : Source 192.168.1.100:12345 - Destination 1.128.0.0:80 - NatIP 1.128.0.0:1024 - Destination 1.128.0.0:8080 - Start Time 2023-04-01T11:00:00Z - Delink Time 2023-04-01T11:05:00Z GMT - Total_bytes_send 102400 - Total_bytes_recv 153600 - Closure Reason Client Reset
-Oct 6 14:03:23 <local0.info> 81.2.69.144 10/06/2014:14:03:23 GMT ns1 0-PPE-0 : TCP NAT_OTHERCONN_DELINK 4471 0 : Source 192.168.1.100:12345 - Destination 1.128.0.0:80 - NatIP 1.128.0.0:1024 - Destination 1.128.0.0:8080 - Start Time 2023-04-01T11:00:00Z - Delink Time 2023-04-01T11:05:00Z GMT - Total_bytes_send 102400 - Total_bytes_recv 153600 - Closure Reason Timeout
```
[the remark this answers is not in the document]
Comment (jrmolin) [reply packages/citrix_adc/data_stream/log/_dev/test/pipeline/test-citrix-waf-native.log:11]: I changed it to match with the timestamp in the syslog header

File: packages/citrix_adc/data_stream/log/_dev/test/pipeline/test-citrix-waf-native.log [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
 Jun 22 19:14:37 <local0.info> 81.2.69.144 12/04/2017:17:21:00 GMT citrix.netscaler.test 0-PPE-1 : SSLLOG SSL_HANDSHAKE_SUCCESS 5743593 0 :  SPCBId 87630 - ClientIP 172.25.184.157 - ClientPort 19849 - VserverServiceIP 10.254.14.94 - VserverServicePort 443 - ClientVersion TLSv1.2 - CipherSuite "RC4-MD5 TLSv1.2 Non-Export 128-bit" - Session Reuse
 Oct 6 14:03:23 <local0.info> 81.2.69.144 10/06/2014:14:03:23 GMT ns1 0-PPE-0 : TCP CONN_DELINK 4471 0 : Source 192.168.10.10:52187 - Vserver 81.2.69.144:80 - NatIP 192.168.10.10:52187 - Destination 81.2.69.144:80 - Delink Time 10/06/2014:14:03:23 GMT - Total_bytes_send 1075 - Total_bytes_recv 352
-Oct 6 14:03:30 <local0.info> 81.2.69.144 10/06/2014:14:03:30 GMT ns1 0-PPE-0 : TCP CONN_TERMINATE 4472 0 : Source 192.168.10.35:80 - Destination 192.168.10.51:35341 - Start Time 10/06/2014:14:02:43 GMT - End Time 10/06/2014:14:03:30 GMT - Total_bytes_send 1 - Total_bytes_recv 1
```
[the remark this answers is not in the document]
Comment (jrmolin) [reply packages/citrix_adc/data_stream/log/_dev/test/pipeline/test-citrix-waf-native.log:4]: I think I meant to copy it somewhere, but I wound up deleting it. I'll put it back .

**Comment by taylor-swanson** [comment]
Superseded by #11698

**Pull request closed without merging**